Bump Devialet to 1.5.7

fwestenberg opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This is an update to make sure Devialet speakers keep working. Because with the update also the JSON response have changed, the fixtures and some tests are updated.

Diff: https://github.com/fwestenberg/devialet/compare/v1.4.5...v1.5.7

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/124395
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description. 

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

fwestenberg committed:
Bump Devialet==1.5.7

zweckj reviewed (dismissed):
- please add (at a minimum) a diff for the changed dependency
- what does "breaking" mean in this case? If you use this with speakers with an old firmware it won't work? If that's the case, please raise an issue telling the user to update.

fwestenberg commented:
> * please add (at a minimum) a diff for the changed dependency
> * what does "breaking" mean in this case? If you use this with speakers with an old firmware it won't work? If that's the case, please raise an issue telling the user to update.

Thanks for your review. A link to the diff is there, right? 
Regarding the "breaking", I noticed a property with probably a changed name. (options->operations). But I'm not sure if this was already an issue before. How would the raising of an issue work? Perhaps there's an example of this?

zweckj commented:
> Thanks for your review. A link to the diff is there, right? 

oh right, can you move this up to the "proposed changes" please?

> Regarding the "breaking", I noticed a property with probably a changed name. (options->operations). But I'm not sure if this was already an issue before. 


what would be the impact of this? Can we find this out before we announce breaking changes?

> How would the raising of an issue work? Perhaps there's an example of this?

https://github.com/home-assistant/core/blob/dev/homeassistant/components/lamarzocco/__init__.py#L132-L142

fwestenberg commented:
> > Thanks for your review. A link to the diff is there, right?
> 
> oh right, can you move this up to the "proposed changes" please?
> 
> > Regarding the "breaking", I noticed a property with probably a changed name. (options->operations). But I'm not sure if this was already an issue before.
> 
> what would be the impact of this? Can we find this out before we announce breaking changes?
> 
> > How would the raising of an issue work? Perhaps there's an example of this?
> 
> https://github.com/home-assistant/core/blob/dev/homeassistant/components/lamarzocco/__init__.py#L132-L142

Moved the diff, also removed the breaking part.

edenhaus reviewed (approved):
Thanks @fwestenberg :+1:
